### ISSUE TITLE
When I type to narrow the search for a local heading all suggestions dissapear

### ISSUE THREAD

**Issue by acristu** [TomasHubelbauer/vscode-markdown-link-suggestions#11]
@TomasHubelbauer

Hi, 

Please see the example below, what am I doing wrong? Expected behavior is that only the suggestion containing "one" would remain in the drop-down after typing "o":
![search_issue](https://user-images.githubusercontent.com/6796822/45025519-61361880-b044-11e8-920d-b353f4fcfb1b.gif)

**Comment by TomasHubelbauer** [comment]
Thank you for the bug report, I can verify this stopped working, probably as a result of me introducing new features while lacking tests for this particular scenario. I have added a new test and will work on fixing this, you can expect a fix to come out in a few days.

**Comment by acristu** [comment]
Thanks.
FYI: also the broken links do not show as problems, might be a regression too...

**Comment by TomasHubelbauer** [comment]
I've started an internal rewrite which aims to enable a use case where the extension will be provide contextual suggestions at any point in the link, which will enable some new features, like suggesting correct relative paths after `/` and skipping suggestions in query strings, but will also fix this issue. Since it's a sizeable rewrite, it will take longer than I initially anticipated. I've decided to do it this way instead of hot fixing your issue in an interest of best utilizing the little time I can dedicate to this extension, since my hotfix would be made redundant by the rewrite anyway. I hope it won't be too much of a bother to wait a bit longer for this, if this breaks your workflow significantly, I can give a second thought to prioritization and do the hotfix for you.
